php/class-media.php

```diff
@@ -1878,34 +1887,39 @@ public function build_cached_meta( $post_id, $key, $single ) {
 	 * @param int          $post_id The attachment ID.
 	 * @param string       $key     The meta key to get.
 	 * @param string|array $data    $the meta data to update.
+	 *
+	 * @return bool
 	 */
 	public function update_post_meta( $post_id, $key, $data ) {
-		$meta_data = wp_get_attachment_metadata( $post_id, true );
-		if ( is_array( $meta_data ) ) {
-			if ( ! isset( $meta_data[ Sync::META_KEYS['cloudinary'] ] ) ) {
-				$meta_data[ Sync::META_KEYS['cloudinary'] ] = array();
-			}
-			$meta_data[ Sync::META_KEYS['cloudinary'] ][ $key ] = $data;
-			wp_update_attachment_metadata( $post_id, $meta_data );
-		} else {
-			// Update core mete data for consistency.
-			update_post_meta( $post_id, $key, $data );
+
+		$meta = $this->get_post_meta( $post_id );
```

Comment:
@DavidCramer 

I think we have a problem here. Because we don't set the `$single` argument, the return will always be in a wrapping array.